### PR TITLE
fix(graphql-model-transformer): fix create mutation when index field is null

### PR DESCRIPTION
**Pull request by lazpavel**
#### Description of changes
- GSI doesnt allow null for the partition key. The solution is to not include that attribute when it is the partition key of a GSI and its null which will ensure the item is never propagated to the GSI which will be the correct behavior and save storage and write cost

#### Description of how you validated changes
- manual testing
- e2e test added

#### Checklist
- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by alharris-at — APPROVED** [review]
Overall lgtm, but I had to go through and do a bit of reading on the VTL template and how the utils/ctx work in general today, so you probably also want to grab a shipit from somebody more authoritative :D

File: packages/amplify-graphql-model-transformer/package.json [hunk 1]
```diff
@@ -44,6 +44,7 @@
         "@aws-cdk/custom-resources": "~1.124.0",
         "@aws-cdk/cx-api": "~1.124.0",
         "@aws-cdk/region-info": "~1.124.0",
+        "amplify-cli-core": "^2.5.0",
```
Comment (sundersc) [inline packages/amplify-graphql-model-transformer/package.json:47]: IMO we shouldn't introduce dependency on CLI core for GraphQL transformers.

File: packages/graphql-transformers-e2e-tests/package.json [hunk 2]
```diff
@@ -22,6 +22,7 @@
     "build-tests": "yarn tsc --build tsconfig.tests.json"
   },
   "dependencies": {
+    "amplify-cli-core": "^2.5.0",
```
Comment (sundersc) [inline packages/graphql-transformers-e2e-tests/package.json:25]: Same comment as above. We should avoid this dependency.

**Review by sundersc — COMMENTED** [review]
LGTM except for the CLI core dependency.

File: packages/graphql-transformers-e2e-tests/package.json (hunk 2 above)
Comment (lazpavel) [reply packages/graphql-transformers-e2e-tests/package.json]: updated

File: packages/amplify-graphql-model-transformer/package.json (hunk 1 above)
Comment (lazpavel) [reply packages/amplify-graphql-model-transformer/package.json]: updated

**Review by sundersc — APPROVED** [review]
LGTM.

**Pull request merged**